autoload/test/javascript/ava.vim

```diff
@@ -0,0 +1,46 @@
+if !exists('g:test#javascript#ava#file_pattern')
+  let g:test#javascript#ava#file_pattern = '\vtests?/.*\.(js|jsx|coffee)$'
+endif
+
+function! test#javascript#ava#test_file(file) abort
+  return a:file =~# g:test#javascript#ava#file_pattern
+    \ && test#javascript#has_package('ava')
+endfunction
+
+function! test#javascript#ava#build_position(type, position) abort
+  if a:type == 'nearest'
+    let name = s:nearest_test(a:position)
+    if !empty(name)
+      let name = '--match='.shellescape(name, 1)
+    endif
+    return [a:position['file'], name]
+  elseif a:type == 'file'
+    return [a:position['file']]
+  else
+    return []
+  endif
+endfunction
+
+function! test#javascript#ava#build_args(args) abort
+  let args = a:args
+
+  if test#base#no_colors()
+    let args = ['--no-colors'] + args
+    let args = args + ['|', 'sed -e "s/
//g"']
```

Comment:
This was copied from Mocha, I would check whether this is needed (as it was specific to Mocha) and/or whether the `ava` runner even supports a `--no-colors` option.

Reply:
The option is slightly different https://github.com/avajs/ava#cli

I will update PR accordingly